python/tests/compat/mantid_test.py

```diff
@@ -4,84 +4,85 @@
 
 import scipp as sc
 import mantid.simpleapi as mantid
+from mantid.api import EventType
 import scipp.compat.mantid as mantidcompat
-# import matplotlib.pyplot as plt
+import numpy as np
 
 
 class TestMantidConversion(unittest.TestCase):
-
     def test_Workspace2D(self):
         # This is from the Mantid system-test data
-        filename = 'CNCS_51936_event.nxs'
+        filename = "CNCS_51936_event.nxs"
         eventWS = mantid.LoadEventNexus(filename)
-        ws = mantid.Rebin(eventWS, -0.001, PreserveEvents=False)
-        d = mantidcompat.to_dataset(ws)
-        print(d)
-        # self.assertEqual(d.dimensions.shape[])
-        # dataset = as_xarray(d[sc.Dim.Position, 1000:2000])
-        # dataset['Value'].plot()
-        # plt.show()
-        # plt.savefig('test.png')
+        ws = mantid.Rebin(eventWS, 10000, PreserveEvents=False)
+        d = mantidcompat.convert_Workspace2D_to_dataset(ws)
+        self.assertEqual(
+            d.labels["run"].value.getProperty("run_start").value,
+            "2012-05-21T15:14:56.279289666",
+        )
 
     def test_EventWorkspace(self):
         # This is from the Mantid system-test data
-        filename = 'CNCS_51936_event.nxs'
+        filename = "CNCS_51936_event.nxs"
         eventWS = mantid.LoadEventNexus(filename)
-        ws = mantid.Rebin(eventWS, -0.001, PreserveEvents=False)
-
-        binned_mantid = mantidcompat.to_dataset(ws)
-
-        tof = sc.Variable(binned_mantid[sc.Coord.Tof])
-        d = mantidcompat.to_dataset(eventWS)
-        binned = sc.histogram(d, tof)
+        ws = mantid.Rebin(eventWS, 10000)
 
-        delta = sc.sum(binned_mantid - binned, sc.Dim.Position)
-        print(delta)
+        binned_mantid = mantidcompat.convert_Workspace2D_to_dataset(ws)
 
-        # ds = as_xarray(delta)
-        # ds['Value'].plot()
-        # plt.savefig('delta.png')
+        target_tof = binned_mantid.coords[sc.Dim.Tof]
+        d = mantidcompat.convert_EventWorkspace_to_dataset(
+            eventWS, False, EventType.TOF
+        )
+        binned = sc.histogram(d, target_tof)
 
-        # ds = as_xarray(sum(binned, sc.Dim.Position))
-        # ds['Value'].plot()
-        # plt.savefig('binned.png')
-
-        # ds = as_xarray(sum(binned_mantid, sc.Dim.Position))
-        # ds['Value'].plot()
-        # plt.savefig('binned_mantid.png')
+        # Following label removal should not be necessary
+        del binned_mantid.labels["run"]
```

Comment:
Removal of PyObjects is because binary operations (transform operations) do not work.  Example:
```python
v = Variable(value=[1,2,3])  
u = Variable(value=[1,2,3])  
u - v
```
Yields:
```RuntimeError: Cannot apply operation to item dtypes PyObject and PyObject.```

It would appear that the fix for this (adding the types to the variant list) is not possible for the same reasons as described in comment above - requires a dependency we may not wish to add.

Reply:
- I think we probably need to add a comparison operator for PyObject somehow, at some point.
- Does my comment above regarding storing run and sample as attributes fix this issue?